includes/forms/class-message-send.php

```diff
@@ -71,4 +71,28 @@ public function __construct( $args = [] ) {
 
 		parent::__construct( $args );
 	}
+
+    /**
+     * Bootstraps form properties.
+     */
+    protected function boot() {
+
+        // Check thread page.
+        if ( hivepress()->request->get_context( 'message_ids' ) ) {
+
+            // Set rendering.
+            $this->attributes['data-render'] = wp_json_encode(
+                [
+                    'url'            => hivepress()->router->get_url( 'message_send_action' ),
+                    'block'          => 'messages',
+                    'event'          => 'submit',
+                    'type'           => 'append',
+                    'fetch_url'      => hivepress()->router->get_url( 'messages_fetch_action' ),
```

Comment:
To avoid adding extra parameters with the need of the fetch_ prefix, please add them directly to the messages container, so the message send form appends a new message on submit:

data-render with url, block, event=submit, type=append

and the messages container basically refreshes itself with an interval:

data-render with url, block (set to itself), interval

Then adding extra params with fetch_ prefix is not needed, the form is not related to the self-refreshing of the messages container at all and it has no trigger events, just an interval